### PR TITLE
Add the auto-release workflow

### PR DESCRIPTION
**Pull request by illright**

### CODE REVIEW

File: .github/workflows/auto_release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Fetch the changelog 📝
+        id: fetch-changelog
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script:
+            const changelog = require('fs').readFileSync('CHANGELOG.md').split('\n');
+            const start = changelog.findIndex(line => line.startsWith('## '));
```
Comment (aabounegm) [inline .github/workflows/auto_release.yml:40]: Wouldn't it be more robust if you use the version returned from the previous step?

File: .github/workflows/auto_release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            const end = changelog.findIndex((line, index) => line.startsWith('## ') && index > start);
+            return changelog.slice(startIndex, endIndex).join('\n');
+
+      - name: Create a release 🔖
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ steps.set-result.outputs.result }}
+          release_name: ${{ steps.set-result.outputs.result }}
```
Comment (aabounegm) [inline .github/workflows/auto_release.yml:50]: Isn't there a better way of providing a name? Perhaps the PR title?

**Review by aabounegm — APPROVED** [review]

File: .github/workflows/auto_release.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          persist-credentials: false
+
+      - name: Fetch the version number 📝
+        id: fetch-version
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script:
+            const package = JSON.parse(require('fs').readFileSync('attractions/package.json'));
```
[the remark this answers is not in the document]
Comment (illright) [reply .github/workflows/auto_release.yml:29]: Oh wow, didn't know Node could import JSON files

File: .github/workflows/auto_release.yml (hunk 1 above)
Comment (illright) [reply .github/workflows/auto_release.yml:40]: It's just a convenience, no big deal. Don't wanna over-engineer it

File: .github/workflows/auto_release.yml (hunk 2 above)
Comment (illright) [reply .github/workflows/auto_release.yml:50]: Release names are overrated, the most informative release name to me is the version number (and that's what we've been doing ourselves for the past couple releases)

**Pull request merged**